File: solved.ac/code/10773.py
import sys
input =sys.stdin.readline
k = int(input())
# k개의 줄에 정수가 한개씩 주어진다. 
ans = []


for i in range(k):

  temp = int(input())

  if temp==0:
    ans.pop()
  else:

    ans.append(temp)

print(sum(ans))

File: solved.ac/code/10773.js
const fs = require("fs")
const input = fs.readFileSync("./dev/stdin", "utf8").trim().split("\n")

// console.log(input)

const [k, ...a] = input

const ans = []

for (let i = 0; i < a.length; i++) {
  if (+a[i] == 0) {
    ans.pop()
  } else {
    ans.push(a[i])
  }
}

let sum = ans.map((v) => +v).reduce((a, x) => a + x, 0)
console.log(sum)
